torchmetrics/utilities/data.py

```diff
@@ -172,12 +172,13 @@ def _stable_1d_sort(x: torch, nb: int = 2049):
     if x.ndim > 1:
         raise ValueError('Stable sort only works on 1d tensors')
     n = x.numel()
-    if n < nb:
-        x_max = x.max()
-        x = torch.cat([x, (x_max + 1) * torch.ones(nb - n, dtype=x.dtype, device=x.device)], 0)
+    if nb is not None:
```

Comment:
IMO we should just remove `nb` option. I don't see when you'd ever want it to be anything other than `2049`. The padding is just a hack to get torch to sort the array in a stable manner (look at the docstring).